nrm_django/grants/forms.py

```diff
@@ -18,10 +26,19 @@ class Meta:
         }
 
     def __init__(self, *args, **kwargs):
-        """A quick init to add some classes on some fields, since we don't need to override the widget."""
+        """
+        A quick init to add some classes on some fields, since we don't need to override the widget,
+        and add a custom field or two.
+        """
         super(GrantForm, self).__init__(*args, **kwargs)
         self.fields["proj_title"].widget.attrs["class"] += " text-wide"
 
+        # If we're modifying an existing instance in a changeform we'll need to set the initial value
+        # for project_category, because we don't have a proper FK from Grant
+        if self.instance:
+            cat_init_cn = Category.objects.get(grant=self.instance).cn
+            self.fields["project_category"].initial = cat_init_cn
```

Comment:
....And here we're attempting to set the initial value of the selectbox for the change_view. But since there are so many duplicates and we used distinct, there is a high likelihood the current choice will not be in the set we created above.

My best idea is to drop the queryset and feed choices to a choicefield, then maybe down here pop and push the choices list to ensure the current value is in there.